### PR TITLE
dask.dataframe.Series.__array_*_ for better use with ufuncs. 

### PR DESCRIPTION
**Pull request by cowlicks**
This "fixes" #580 

However it does not do the right thing. The right thing would be for `np.ufunc(dask.Series)` to return a `dask.Series` now it forces computation and returns a pandas dataframe.

We could eventually do that when it `__numpy_ufunc__` is included in NumPy, which might happen in numpy 1.10 or 1.11

### CODE REVIEW

**Comment by cowlicks** [comment]
Interesting my tests pass on pandas 0.16.1 but not 0.16.2

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -512,6 +512,18 @@ def __repr__(self):
...
+    def __array__(self, dtype=None, **kwargs):
+        x = np.array(self.compute())
+        if dtype and x.dtype != dtype:
+            x = x.astype(dtype)
+        return x
+
+    def __array_wrap__(self, array, context=None):
+        return pd.Series(array, name=self.name)
+
+    def __array_prepare__(self, array, context=None):
```
Comment (shoyer) [inline dask/dataframe/core.py]: I don't think you need `__array_prepare__` in this case

Comment (cowlicks) [inline dask/dataframe/core.py]: I didn't. Thanks.

**Comment by cowlicks** [comment]
We could also raise an exception when people call ufuncs on dask.arrays/dataframes. Would that be preferable to silently forcing computation?

**Comment by mrocklin** [comment]
Where is the failure in pandas 0.16.1.  

I would be for raising a warning.

**Comment by mrocklin** [comment]
Note that the underlying problem of #580 is that we don't have a way to say 

    log(myseries)

for any function `log`.  Arguably the `dask.array` "ufuncs" should operate smoothly on `dask.dataframe` `Series` objects.  That's probably the full solution in the near term.

**Comment by cowlicks** [comment]
@mrocklin the in 0.16.1 the way I was creating the output series [here](https://github.com/ContinuumIO/dask/pull/582/files#diff-492da7893fa5fa6ad3a6ef6cdef985f2R522) worked without the `name` kwarg because the name was not being checked in `assert_series_equal`. In 0.16.2 `assert_series_equal` started to check names.

**Comment by mrocklin** [comment]
OK.  Merging this.  If you feel like some adventure it'd still be good to get the dask.array ufuncs to work on dask.dataframe objects.  I think that the best way to do this is likely some sort of protocol so that dask.dataframe (or dask.bag) logic doesn't find its way into dask.array

**Pull request merged**